packages/flutter/test/widgets/selectable_region_test.dart

```diff
@@ -4440,6 +4440,310 @@ void main() {
+    expect((activeSelections[0].content as TextSpan).text, 'How are you?');
+    expect(activeSelections[1].startOffset, 0);
+    expect(activeSelections[1].endOffset, 14);
+    expect((activeSelections[1].content as TextSpan).text, isNotNull);
+    expect((activeSelections[1].content as TextSpan).text, 'Good, and you?');
+    expect(activeSelections[2].startOffset, 0);
+    expect(activeSelections[2].endOffset, 10);
+    expect((activeSelections[2].content as TextSpan).text, isNotNull);
+    expect((activeSelections[2].content as TextSpan).text, 'Fine, thank you.');
+    await mouseGesture.up();
+    await tester.pump();
+    expect(activeSelections, isNotEmpty);
+    expect(activeSelections.length, 3);
+
+    // Collapsed selection.
+    await mouseGesture.down(textOffsetToPosition(paragraph1, 3));
+    await tester.pump();
+    await mouseGesture.up();
+    await tester.pumpAndSettle(kDoubleTapTimeout);
+    expect(activeSelections, isNotEmpty);
+    expect(activeSelections.length, 1);
+    expect(activeSelections[0].startOffset, 3);
+    expect(activeSelections[0].endOffset, 3);
+    expect((activeSelections[0].content as TextSpan).text, isNotNull);
+    expect((activeSelections[0].content as TextSpan).text, 'How are you?');
+
+    // Backwards selection.
+    await mouseGesture.down(textOffsetToPosition(paragraph3, 4));
+    await tester.pump();
+    await mouseGesture.moveTo(textOffsetToPosition(paragraph1, 0));
+    await tester.pumpAndSettle();
+    expect(activeSelections, isNotEmpty);
+    expect(activeSelections.length, 3);
+    expect(activeSelections[0].startOffset, 12);
+    expect(activeSelections[0].endOffset, 0);
+    expect((activeSelections[0].content as TextSpan).text, isNotNull);
+    expect((activeSelections[0].content as TextSpan).text, 'How are you?');
+    expect(activeSelections[1].startOffset, 14);
+    expect(activeSelections[1].endOffset, 0);
+    expect((activeSelections[1].content as TextSpan).text, isNotNull);
+    expect((activeSelections[1].content as TextSpan).text, 'Good, and you?');
+    expect(activeSelections[2].startOffset, 4);
+    expect(activeSelections[2].endOffset, 0);
+    expect((activeSelections[2].content as TextSpan).text, isNotNull);
+    expect((activeSelections[2].content as TextSpan).text, 'Fine, thank you.');
+    await mouseGesture.up();
+    await tester.pump();
+    expect(activeSelections, isNotEmpty);
+    expect(activeSelections.length, 3);
+
+    // Collapsed selection.
+    await mouseGesture.down(textOffsetToPosition(paragraph1, 0));
+    await tester.pumpAndSettle();
+    await mouseGesture.up();
+    await tester.pumpAndSettle(kDoubleTapTimeout);
+    expect(activeSelections, isNotEmpty);
+    expect(activeSelections.length, 1);
+    expect(activeSelections[0].startOffset, 0);
+    expect(activeSelections[0].endOffset, 0);
+    expect((activeSelections[0].content as TextSpan).text, isNotNull);
+    expect((activeSelections[0].content as TextSpan).text, 'How are you?');
+  });
```

Comment:
Good tests, they help me understand how this all works in addition to covering the code :+1: